torch/nn/_functions/rnn.py

```diff
@@ -300,14 +302,18 @@ def backward_extended(self, grad_output, grad_hy):
             grad_hx)
 
         if any(self.needs_input_grad[1:]):
-            grad_weight = [tuple(w.new().resize_as_(w) for w in layer_weight) for layer_weight in weight]
+            grad_weight = [tuple(w.new().resize_as_(w) if w is not None else None for w in layer_weight)
+                           for layer_weight in weight]
             cudnn.rnn.backward_weight(
                 self,
                 input,
                 hx,
                 output,
                 weight,
                 grad_weight)
+            if self.skip_input:
+                grad_weight = [tuple(w for w in layer_grad_weight if w is not None)
+                               for layer_grad_weight in grad_weight]
```

Comment:
I think you might need to filter out the grads for None weights in the `else` clause too.